internal/pkg/cli/pipeline_init.go

```diff
@@ -359,24 +355,44 @@ func relPath(fullPath string) string {
 	return relPath
 }
 
-// TODO: Nice-to-have: have an opts.listRemoteRepos() method that execs out to `git remote -v` and parse repo name to offer select menu
-func (opts *InitPipelineOpts) selectGitHubRepo() error {
-	repo, err := opts.prompt.Get(
-		pipelineEnterGitHubRepoPrompt,
-		fmt.Sprintf(`The GitHub repository linked to your workspace. Pushing to this repository will trigger your pipeline build stage. Please enter full repository URL, e.g. "https://github.com/myCompany/myRepo", or the owner/rep, e.g. "myCompany/myRepo"`),
-		validateGitHubRepo,
+func (opts *InitPipelineOpts) getGitHubOwner() error {
+	owner, err := opts.prompt.Get(
+		pipelineEnterGitHubOwnerPrompt,
+		pipelineEnterGitHubOwnerHelpPrompt,
+		nil,
 	)
+	if err != nil {
+		return fmt.Errorf("fail to get GitHub owner name: %w", err)
+	}
+	opts.GitHubOwner = owner
+	return nil
+}
 
-	// Remove ".git" at the end of the Github repo URL.
-	repoNameSplit := strings.Split(repo, ".")
-	if repoNameSplit[len(repoNameSplit)-1] == "git" {
-		repo = strings.Join(repoNameSplit[:len(repoNameSplit)-1], ".")
+func (opts *InitPipelineOpts) listGitHubRepo() ([]string, error) {
+	repos, _, err := opts.repo.List(opts.context, opts.GitHubOwner, &github.RepositoryListOptions{})
```

Comment:
Also, we should probably check for the case when `opts.GitHubOwner` has not been set, just in case `listGitHubRepo` is called before `getGitHubOwner`.

Same principal applies for `getGitHubBranch` below

Reply:
Yeah, we need to add pagination to this. And I'm going to remove listing branches so god thanks I don't need to add pagination for that.